packages/repository/src/__tests__/acceptance/has-one.relation.acceptance.ts

```diff
@@ -115,6 +115,61 @@ describe('hasOne relation', () => {
     ).to.be.rejectedWith(EntityNotFoundError);
   });
 
+  it('can PATCH hasOne instances', async () => {
+    //    existingCustomerId = 11;
+    const address = await controller.createCustomerAddress(existingCustomerId, {
+      street: '1 Amedee Bonnet',
+      zipcode: '69740',
+      city: 'Genas',
+      province: 'Rhone',
+    });
+
+    // const persisted = await addressRepo.findById(address.zipcode);
+    // expect(persisted.toObject()).to.deepEqual(address.toObject());
+
+    const patchObject = {city: 'Lyon-Genas'};
+    const arePatched = await controller.patchCustomerAddress(
+      existingCustomerId,
+      patchObject,
+    );
+
+    expect(arePatched).to.deepEqual({count: 1});
+    const patchedData = await addressRepo.findById(address.zipcode);
+    expect(patchedData.city).to.containEql(address.city);
+  });
+
+  it('can throws error when query tries to change the foreignKey', async () => {
+    try {
+      await expect(
+        controller.patchCustomerAddress(existingCustomerId, {
+          customerId: existingCustomerId + 1,
+        }),
+      ).to.be.rejectedWith(/Property "customerId" cannot be changed!/);
+    } catch (err) {}
+  });
+
+  it('can DELETE hasOne relation instances', async () => {
+    //    existingCustomerId = 12;
+    const address = await controller.createCustomerAddress(existingCustomerId, {
+      street: '1 Amedee Bonnet',
+      zipcode: '69740',
+      city: 'Genas',
+      province: 'Rhone',
+    });
+
+    const newData = await addressRepo.findById(address.zipcode);
+    expect(newData.city).to.containEql(address.city);
```

Comment:
There is no need to verify that `createCustomerAddress` works correctly, we have already verified that in other tests. Please remove these two lines.

Reply:
Ok